### ISSUE TITLE
Bugs with Megasat SAT>IP Server 2

### ISSUE THREAD

**Issue by jmachowinski** [rofafor/vdr-plugin-satip#6]
Hi,
I tried your plugin with the Megasat SAT>IP Server 2
and discovered some bugs.

The Megasat SAT>IP Server 2 identifies itself as 
'Dvbs2-4' but the cDvbTransponderParameters it
is reported as dvbs (not dvbs2).

SatIp Spec 1.2.2 specifies, that 
Roll-Off, Modulation Type and Pilot Tones
shall be given in the client request. If they are
not given, the Megasat device reports a 'Internal Server Error' 
in the Setup phase.

There is a bug with the rtsp session id. The first character of
the session id gets lost somewhere. It is extracted correctly   
in the cSatipRtsp::HeaderCallback but somehow it gets lost 
in the curl calls. Note, if I set it correctly in the Tuner: SetSessionTimeout
using  rtspM.SetSession(SkipZeroes(*sessionM)); it works, but
I get CURL errors in the syslog.

For now I only have hacky workarounds, for the 1.03 version.
If I come up with a real patch, I will report back.

**Comment by rofafor** [comment]
Could you provide me Megasat's XML files? I need those for fixing the reported discovery and session problems.

**Comment by jmachowinski** [comment]
I traced the session problem down to curl 7.35. 7.36 ist working.

The XML is :
<?xml version="1.0"?><root xmlns= "urn:schemas-upnp-org:device-1-0" configId="45298"><specVersion><major>1</major><minor>1</minor></specVersion><device><deviceType>urn:ses-com:device:SatIPServer:1</deviceType><friendlyName>Sat>IP Server</friendlyName><manufacturer>Megasat Werke GmbH</manufacturer><modelName>MEG-4000</modelName><modelNumber>1.0</modelNumber><serialNumber>1S81A31231000007</serialNumber><UDN>uuid:ccc06f24-79c8-11e4-00b1-b8871e00af6c</UDN><iconList><icon><mimetype>image/png</mimetype><width>48</width><height>48</height><depth>24</depth><url>/icons/sm.png</url></icon><icon><mimetype>image/png</mimetype><width>120</width><height>120</height><depth>24</depth><url>/icons/lr.png</url></icon><icon><mimetype>image/jpeg</mimetype><width>48</width><height>48</height><depth>24</depth><url>/icons/sm.jpeg</url></icon><icon><mimetype>image/jpeg</mimetype><width>120</width><height>120</height><depth>24</depth><url>/icons/lr.jpeg</url></icon></iconList><presentationURL>/index.html</presentationURL><satip:X_SATIPCAP xmlns:satip="urn:ses-com:satip">DVBS2-4</satip:X_SATIPCAP></device></root>

**Comment by rofafor** [comment]
Read the requirements before adding any bug reports.

**Comment by jmachowinski** [comment]
I installed the plugin via easyvdr. The dependency is broken in their repo...
Also I would recommend, to change the build in compile warning for a 
to low curl version to a compile error.

The bug with the Roll-Off, Modulation Type and Pilot Tones still exists.

For the DVB-S vs DVB-S2 thing, I figured out that there is no bug, as
the given parameter corresponds to the channel, and not the the
capabilities of the SatIp Server. Anyway, the channel passed during
the RTSP Setup is a DVB-S one. As the parameters in setup differ 
in reaction to the given channel, this might be bad.
